File: binary.py
# loop menu
while True:
    # menu
    print("MENU:")
    print("1. to Binary")
    print("2. from binary")
    print("3. quit")
    choice = input("enter 1-3: ")
    # to
    if choice == '1':
        print("translating to binary")
        dec = int(input("decimal input: "))
        if dec == 0:
            dec == ("0b0")
        output = bin(dec)
        final = ("the equivelant is " + str(output[2:]))
        print((final))
        print("quit: y")
        print("don't: n")
        exit_prompt = input("answer: ")
        if exit_prompt == "y":
            exit
            break
        elif exit_prompt == "n":
            pass 
    # from
    elif choice == '2':
        print("translating from binary")
        binary_start = input("enter 1's and 0's: ")
        dec_end = int(binary_start, 2)
        dec_fin = ("the equivelant is " + str(dec_end))
        print(dec_fin)
        print("quit: y")
        print("don't: n")
        exit_prompt = input("answer: ")
        if exit_prompt == "y":
            exit
            break
        elif exit_prompt == "n":
            pass 
    # invalid & exit
    elif choice == '3':
        print("confirm: y")
        print("don't: n")
        exit_prompt = input("answer: ")
        if exit_prompt == "y":
            exit
            break
        elif exit_prompt == "n":
            pass 
    else:
        print("invalid, goodbye...")
        exit
        break

File: binary.go
package main

//dont touch anything it works
import (
	"fmt"
	"os"
	"strconv"
)

func binaryToDecimal(bin int) {
	binaryStr := strconv.FormatInt(int64(bin), 10)
	decimal, err := strconv.ParseInt(binaryStr, 2, 64)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	fmt.Print("value is: ")
	fmt.Println(decimal)
}

func decimalToBinary(dec int) string {
	return strconv.FormatInt(int64(dec), 2)
}
func main() {
	for {
		fmt.Println("1. to binary")
		fmt.Println("2. from binary")
		fmt.Println("3. quit")
		var choice int
		fmt.Println("enter 1-3: ")
		_, err := fmt.Scanln(&choice)
		if err != nil {
			fmt.Println("error")
			return
		}
		// to
		if choice == 1 {
			var dec int
			fmt.Println("you selected to binary")
			fmt.Println("decimal input")
			_, err = fmt.Scanln(&dec)
			if err != nil {
				fmt.Println("error")
				return
			}
			bin := decimalToBinary(dec)
			fmt.Print("value is: ")
			fmt.Println(bin)
		} else if choice == 2 {
			var bin int
			fmt.Println("you selected from binary")
			fmt.Println("binary input")
			_, err := fmt.Scanln(&bin)
			if err != nil {
				fmt.Println("error")
				return
			}
			(binaryToDecimal(bin))
		} else if choice == 3 {
			fmt.Println("exiting")
			os.Exit(0)
		} else {
			fmt.Println("invalid. goodbye")
			os.Exit(0)
		}
	}
}
